File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/sql/OrdersPerCity.scala
package io.devorbit.sparklabs.batch.sales.sql

object OrdersPerCity extends App {
  SqlUtils.sql(
    """
      |SELECT c.city, COUNT(*) AS total_orders_per_city
      |FROM orders o
      |JOIN customers c ON o.customer_id = c.customer_id
      |GROUP BY c.city
      |ORDER BY total_orders_per_city DESC
      |""".stripMargin
  ).show(false)
}

File: python/sparklabs/batch/sales/sql/orders_per_city.py
from sparklabs.batch.sales.sql import utils


def run() -> None:
    utils.sql(
        """
        SELECT c.city, COUNT(*) AS total_orders_per_city
        FROM orders o
        JOIN customers c ON o.customer_id = c.customer_id
        GROUP BY c.city
        ORDER BY total_orders_per_city DESC
        """
    ).show(truncate=False)


if __name__ == "__main__":
    run()
